hs_swat_modelinstance/models.py

```diff
@@ -0,0 +1,550 @@
+__author__ = 'Mohamed Morsy'
+from django.contrib.contenttypes import generic
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from django.contrib.auth.models import User, Group
+from django.db import models
+from mezzanine.pages.models import Page, RichText
+from mezzanine.core.models import Ownable
+from mezzanine.pages.page_processors import processor_for
+from hs_core.models import AbstractResource, resource_processor, CoreMetaData, AbstractMetaDataElement
+
+
+
+
+# extended metadata elements for SWAT Model Instance resource type
+class ModelOutput(AbstractMetaDataElement):
+    term = 'ModelOutput'
+    includes_output = models.BooleanField(default=False)
+
+    @classmethod
+    def create(cls, **kwargs):
+        return ModelOutput.objects.create(**kwargs)
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        model_output = ModelOutput.objects.get(id=element_id)
+        if model_output:
+            for key, value in kwargs.iteritems():
+                setattr(model_output, key, value)
+
+            model_output.save()
+        else:
+            raise ObjectDoesNotExist("No ModelOutput element was found for the provided id:%s" % kwargs['id'])
```

Comment:
1) Using `ModelOutput.objects.update()` would be more efficient. One query instead of two.
2) There are several implementations of this for different models that are all identical.